core/client/fs/src/main/java/alluxio/client/file/UfsBaseFileSystem.java

```diff
@@ -0,0 +1,425 @@
+        ufsOptions.setRecursive(options.getRecursive());
+      }
+      UfsStatus[] ufsStatuses = mUfs.get().listStatus(path.getPath(), ufsOptions);
+      if (ufsStatuses == null || ufsStatuses.length == 0) {
+        return Collections.emptyList();
+      }
+      return Arrays.stream(ufsStatuses).map(this::transformStatus).collect(Collectors.toList());
+    });
+  }
+
+  @Override
+  public void iterateStatus(AlluxioURI path, final ListStatusPOptions options,
+      Consumer<? super URIStatus> action) {
+    call(() -> {
+      ListOptions ufsOptions = ListOptions.defaults();
+      if (options.hasRecursive()) {
+        ufsOptions.setRecursive(options.getRecursive());
+      }
+      UfsStatus[] ufsStatuses = mUfs.get().listStatus(path.getPath(), ufsOptions);
+      if (ufsStatuses == null || ufsStatuses.length == 0) {
+        return;
+      }
+      Arrays.stream(ufsStatuses).map(this::transformStatus).forEach(action);
+    });
+  }
+
+  @Override
+  public ListStatusPartialResult listStatusPartial(
+      AlluxioURI path, final ListStatusPartialPOptions options) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void loadMetadata(AlluxioURI path, final ListStatusPOptions options) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void mount(AlluxioURI alluxioPath, AlluxioURI ufsPath, final MountPOptions options) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void updateMount(AlluxioURI alluxioPath, final MountPOptions options) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public Map<String, MountPointInfo> getMountTable(boolean checkUfs) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public List<SyncPointInfo> getSyncPathList() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void persist(final AlluxioURI path, final ScheduleAsyncPersistencePOptions options) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public FileInStream openFile(AlluxioURI path, OpenFilePOptions options) {
+    return openFile(getStatus(path), options);
+  }
+
+  @Override
+  public FileInStream openFile(URIStatus status, OpenFilePOptions options) {
+    return callWithReturn(() -> {
+      // TODO(lu) deal with other options e.g. maxUfsReadConcurrency
+      return new UfsFileInStream(mUfs.get().open(status.getPath()),
+          status.getLength());
+    });
+  }
+
+  @Override
+  public void rename(AlluxioURI src, AlluxioURI dst, RenamePOptions options) {
+    call(() -> {
+      String srcPath = src.getPath();
+      String dstPath = dst.getPath();
+      boolean renamed;
+      if (mUfs.get().isFile(srcPath)) {
+        renamed = mUfs.get().renameFile(srcPath, dstPath);
+      } else {
+        renamed = mUfs.get().renameDirectory(srcPath, dstPath);
+      }
+      if (!renamed) {
+        throw new AlluxioRuntimeException(
+            Status.UNKNOWN,
```

Comment:
is it possible to get the error message from ufs?

Reply:
Ideally UFS should directly throw AlluxioRuntimeExceptions because there may be different reasons, let me try to synchronize some common error message in this PR
e.g. source does not exist or is directory

Reply:
One future change is in the TODO, let UFS directly throw detailed runtime exceptions instead of returning a boolean with the only error message in the logs.